inc/question.class.php

```diff
@@ -1186,4 +1186,93 @@ public static function dropdownForForm($formId, $crit, $name, $value) {
       $items = $question->getQuestionsFromFormBySection($formId, $crit);
       Dropdown::showFromArray($name, $items, []);
    }
+
+   /**
+    * Get target subdata (conditions, regexes or ranges)
+    *
+    * @param string $target target table
+    * @param id|array $condition condition
+    * @return array
+    */
+   public static function getQuestionSubdata($target, $condition) {
+      global $DB;
+
+      $validTargets = [
+         \PluginFormcreatorQuestion_Condition::getTable(),
+         \PluginFormcreatorQuestionRegex::getTable(),
+         \PluginFormcreatorQuestionRange::getTable(),
+      ];
+
+      if (array_search($target, $validTargets) === false) {
+         throw new \InvalidArgumentException("Invalid target ('$target')");
+      }
+
+      return iterator_to_array($DB->request([
+         'FROM' => $target,
+         'WHERE' => [
+            "plugin_formcreator_questions_id" => $condition
```

Comment:
A suggestion to have a more powerful code : maybe $condition should be an array of the complete WHERE statement of the query. See CommonDBTM::getFromDBByCrit().

I would write the array of the request as the following. 
```php
      return iterator_to_array($DB->request([
         'FROM' => $target,
         'WHERE' => $condition
      ]));  
```

and adapt all calls of the method. This makes the name $condition more appropriate also.

Reply:
I think the method was badly named, it is only used to avoid repetitive queries in getFullData (and thus I do not wish to rewrite the full condition), I've made some changes and it should be more straightforward now.